File: android/TodokTodok/src/main/java/todoktodok/backend/book/application/dto/response/BookResponse.java
package todoktodok.backend.book.application.dto.response;

import todoktodok.backend.book.domain.Book;

public record BookResponse(
        Long bookId,
        String bookTitle,
        String bookAuthor,
        String bookImage
) {

    public BookResponse(Book book) {
        this(
                book.getId(),
                book.getTitle(),
                book.getAuthor(),
                book.getImage()
        );
    }
}

File: android/TodokTodok/app/src/main/java/com/team/todoktodok/data/network/response/discussion/BookResponse.kt
package com.team.todoktodok.data.network.response.discussion

import com.team.domain.model.Book
import kotlinx.serialization.Serializable

@Serializable
data class BookResponse(
    val bookAuthor: String,
    val bookId: Long,
    val bookImage: String,
    val bookTitle: String,
    val bookPublisher: String = "",
    val bookSummary: String = "",
)

fun BookResponse.toDomain() =
    Book(
        id = bookId,
        title = bookTitle,
        author = bookAuthor,
        image = bookImage,
        publisher = bookPublisher,
        summary = bookSummary,
    )
